js/runner.go

```diff
@@ -432,6 +429,22 @@ func (r *Runner) HandleSummary(ctx context.Context, summary *lib.Summary) (map[s
 	return getSummaryResult(rawResult)
 }
 
+func (r *Runner) checkDeadline(ctx context.Context, name string, result goja.Value, err error) error {
+	// deadline is reached so we have timeouted but this might've not been registered correctly
+	if deadline, ok := ctx.Deadline(); ok && time.Now().After(deadline) {
```

Comment:
Seems like we could easily reduce number of nested if here, which improves the method readability

Reply:
[0e0cc40](https://github.com/grafana/k6/pull/3393/commits/0e0cc401986201d3f238f7e76db48bab0a8d5dc7)